src/mscorlib/src/System/Diagnostics/Stacktrace.cs

```diff
@@ -74,10 +98,81 @@ public StackFrameHelper(bool fNeedFileLineColInfo, Thread target)
             // limit in the future, then we should expose it in the managed API so applications can 
             // override it.
             iFrameCount = 0;
+        }
+
+        //
+        // Initializes the stack trace helper. If fNeedFileInfo is true, initializes rgFilename, 
+        // rgiLineNumber and rgiColumnNumber fields using the portable PDB reader if not already
+        // done by GetStackFramesInternal (on Windows for old PDB format).
+        //
+        internal void InitializeSourceInfo(int iSkip, bool fNeedFileInfo, Exception exception)
+        {
+            StackTrace.GetStackFramesInternal(this, iSkip, fNeedFileInfo, exception);
+
+            if (!fNeedFileInfo)
+                return;
+
+            // Check if this function is being reentered because of an exception in the code below
+            if (t_reentrancy > 0)
+                return;
+
+            t_reentrancy++;
+            try
+            {
+                if (s_symbolsMethodInfo == null)
+                {
+                    Assembly metadataAssembly = Assembly.Load("System.Diagnostics.StackTrace, Version=1.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a");
+                    if (metadataAssembly == null)
+                        return;
+
+                    s_symbolsType = metadataAssembly.GetType("System.Diagnostics.StackTraceSymbols");
```

Comment:
Same here.

You may want to combine the both calls into:

`Type.GetType("System.Diagnostics.StackTraceSymbols, System.Diagnostics.StackTrace...", throwOnError: false)`